framework/src/loops/ComputeMaterialsObjectThread.C

```diff
@@ -171,7 +171,15 @@ ComputeMaterialsObjectThread::onInternalSide(const Elem * elem, unsigned int sid
 {
   if (_need_internal_side_material)
   {
-    _assembly[_tid]->reinit(elem, side);
+    const Elem * neighbor = elem->neighbor_ptr(side);
+    const dof_id_type elem_id = elem->id(), neighbor_id = neighbor->id();
+    const bool should_continue =
```

Comment:
It is a bit hard to understand this logic. Could you add comments ?

Reply:
Can we just add this logic as a utility somewhere if we use it multiple times? I think we use the same logic in DG? Maybe something on MooseMesh?